### PR TITLE
fix(lsp): regression - formatting was broken on windows

### LINKED ISSUE #21971: regression: 1.39.3 and 1.39.4 breaks the denoland.vscode-deno vscode formatter on windows
Version: Deno 1.39.3, 1.39.4
Seems to be affecting Windows only. Mac works fine. Can't reproduce on Linux.

Neither the formatting on save nor manual request to format work on the affected versions.

`deno upgrade --version 1.39.2` restores the formatter (after rebooting VSCode), no need to downgrade the vscode extension which is [3.31.0](https://github.com/denoland/vscode_deno/compare/3.30.0...3.31.0) / 2023.12.28
Formatting on save or manual request to format works as expected.

My local `.vscode` `settings.json` settings are:

```json
  "javascript.format.enable": false,
  "typescript.format.enable": false,
  "editor.defaultFormatter": "denoland.vscode-deno",
  "editor.formatOnSave": true,
  "[typescriptreact]": {
    "editor.defaultFormatter": "denoland.vscode-deno"
  },
  "[typescript]": {
    "editor.defaultFormatter": "denoland.vscode-deno"
  },
  "[javascriptreact]": {
    "editor.defaultFormatter": "denoland.vscode-deno"
  },
  "[javascript]": {
    "editor.defaultFormatter": "denoland.vscode-deno"
  },
  "[json]": {
    "editor.defaultFormatter": "denoland.vscode-deno"
  },
  "[jsonc]": {
    "editor.defaultFormatter": "denoland.vscode-deno"
  },
  "editor.codeActionsOnSave": {
    "source.organizeImports": "never",
    "source.fixAll": "never"
  },
  "[css]": {
    "editor.defaultFormatter": "vscode.css-language-features"
  }
```

My global settings has nothing regarding formatting.

### PR DESCRIPTION
**Pull request by dsherret**
~~Waiting on: https://github.com/denoland/deno_config/pull/31~~

Closes #21971
Closes https://github.com/denoland/vscode_deno/issues/1029

### CODE REVIEW

File: cli/lsp/language_server.rs [hunk 1]
```diff
@@ -1046,20 +1052,18 @@ impl Inner {
     self.fmt_options = Default::default();
     self.lint_options = Default::default();
     if let Some(config_file) = self.get_config_file()? {
-      // this doesn't need to be an actual directory because flags is specified as `None`
```
Comment (dsherret) [inline cli/lsp/language_server.rs:1049]: Lies I wrote.

File: cli/args/mod.rs [hunk 2]
```diff
@@ -1666,15 +1665,7 @@ fn resolve_files(
...
-    include: {
-      let files = match maybe_files_config.include {
-        Some(include) => include,
-        None => PathOrPatternSet::new(vec![PathOrPattern::Path(
-          initial_cwd.to_path_buf(),
-        )]),
-      };
-      Some(files)
-    },
+    include: maybe_files_config.include,
```
Comment (dsherret) [inline cli/args/mod.rs:1668]: This reverts this change I made. It shouldn't have been done.

File: cli/args/mod.rs [hunk 3]
```diff
@@ -244,14 +243,22 @@ impl BenchOptions {
...
   pub options: FmtOptionsConfig,
   pub files: FilePatterns,
 }
 
 impl FmtOptions {
+  pub fn new_with_base(base: PathBuf) -> Self {
+    Self {
+      check: false,
+      options: FmtOptionsConfig::default(),
+      files: FilePatterns::new_with_base(base),
```
Comment (dsherret) [inline cli/args/mod.rs:258]: See explanation for this `base` here: https://github.com/denoland/deno_config/pull/31#discussion_r1456423798

**Review by bartlomieju — APPROVED** [review]
LGTM, no comments

**Review by nayeemrmn — APPROVED** [review]

**Pull request merged**